packages/nuxt/src/core/plugins/detect-component-usage.ts

```diff
@@ -0,0 +1,43 @@
+import { createUnplugin } from 'unplugin'
+import { join, resolve } from 'pathe'
+import { updateTemplates } from '@nuxt/kit'
+import { distDir } from '../../dirs'
+
+interface DetectComponentUsageOptions {
+  rootDir: string
+  exclude?: Array<RegExp | string>
+  include?: Array<RegExp | string>
+  detectedComponents: Set<string>
+}
+
+export const DetectComponentUsagePlugin = (options: DetectComponentUsageOptions) => createUnplugin(() => {
+  const importersToExclude = options?.exclude || []
+  const importersToInclude = options?.include || []
+
+  const detectComponentUsagePatterns: Array<[importPattern: string | RegExp, name: string]> = [
+    [resolve(distDir, 'pages/runtime/page'), 'NuxtPage'],
+    [resolve(distDir, 'app/components/nuxt-layout'), 'NuxtLayout']
+  ]
+
+  return {
+    name: 'nuxt:detect-component-usage',
+    enforce: 'pre',
+    resolveId (id, importer) {
+      if (!importer) { return }
+      if (id[0] === '.') {
+        id = join(importer, '..', id)
+      }
```

Comment:
This logic doesn't quite look right... What about directories beginning with `.` - and if it's a relative path surely we could just do `join(importer, id)` rather than inserting an extra level up?